modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/utils/fetchUtil.ts

```diff
@@ -69,7 +69,7 @@ export async function getAllItems<T>({
 			page: number;
 		}>({
 			input: filter
-				? `${url}?filter=${filter}&?page=${currentPage}`
+				? `${url}?filter=${filter}&page=${currentPage}`
```

Comment:
Try to test this:
- Delete all the unnecessary code (do while) and just only set a `pageSize=-1`. I think that the behavior will be the same because right now Pagination has a Configuration and the default value is 500. The second point will be use less code.
- If my suggestion works well for APIEndpoint

Reply:
Makes sense, and it works, thanks! :)